lib/async_hooks.js

```diff
@@ -204,6 +205,105 @@ class AsyncResource {
   }
 }
 
+const locals = [];
+const asyncContextHook = createHook({
+  init(asyncId, type, triggerAsyncId, resource) {
+    for (let i = 0; i < locals.length; ++i) {
+      locals[i]._propagate(resource, executionAsyncResource());
+    }
+  }
+});
+
+class AsyncLocalStorage {
+  constructor() {
+    this.kResourceStore = Symbol('kResourceStore');
+  }
+
+  /**
+   * Propagate the context from a parent resource to a child one.
+   */
+  _propagate(resource, triggerResource) {
+    const store = triggerResource[this.kResourceStore];
+    if (store) {
+      resource[this.kResourceStore] = store;
```

Comment:
```suggestion
      resource[this.kResourceStore] = store;
```

I wonder if we shouldn't use a Map or WeakMap with `resource` as the key instead of putting the store on the resource itself?

Reply:
I have been experiencing about it and it is mostly to reduce performance overhead. The `Symbol` track is slightly more performant (https://www.measurethat.net/Benchmarks/ShowResult/93128). 
Do you see any issue we could meet with the `Symbol` approach?

Reply:
> Do you see any issue we could meet with the Symbol approach?

No